eventmembershipsignup.php

```diff
@@ -119,30 +201,24 @@ function eventmembershipsignup_civicrm_pre($op, $objectName, $id, &$params) {
   AND p.contact_id = c.contact_id
 WHERE c.id = %1
   AND os.entity_table = 'Event'
+UNION
+SELECT pp.participant_id as participant_id, part.status_id as participant_status_id
+FROM `civicrm_participant_payment` pp
+LEFT JOIN civicrm_participant part
+  ON part.id = pp.participant_id
+WHERE pp.contribution_id = %1
```

Comment:
Why do you do this union?  This would presumably pull in the participant records that were actually tied to the contribution.  We shouldn't be touching those.